src/BankAccount.java

```diff
@@ -0,0 +1,166 @@
+import java.util.HashMap;
+
+class Account {
```

Comment:
This is set up sort of like the node class in project 6. This is fine, but I might recommend an inner interface for part 3. See Map.Pair for what I mean. Interestingly, you could also just use a map, where the keys are the values you want to store, and the values are the data. However, that won't work as well with the types of data you're looking to store.